src/WindowInfoProvider/index.js

```diff
@@ -1,115 +1,93 @@
-import React, { Component } from 'react';
+import React, { useState, useEffect } from 'react';
 import PropTypes from 'prop-types';
 import WindowInfoContext from './context';
 
-class WindowInfoProvider extends Component {
-  constructor(props) {
-    super(props);
-
-    this.state = {
-      animationScheduled: false,
-      width: 0,
-      height: 0,
-      breakpoints: {
-        xs: false,
-        s: false,
-        m: false,
-        l: false,
-        xl: false,
-      },
-      eventsFired: 0,
-    };
-  }
-
-  componentDidMount() {
-    window.addEventListener('resize', this.requestAnimation);
-    window.addEventListener('orientationchange', this.updateWindowInfoWithTimeout);
-    this.updateWindowInfo();
-  }
-
-  componentWillUnmount() {
-    window.removeEventListener('resize', this.requestAnimation);
-    window.removeEventListener('orientationchange', this.updateWindowInfoWithTimeout);
-  }
-
-  updateWindowInfoWithTimeout = () => {
-    setTimeout(() => {
-      this.requestAnimation();
-    }, 500);
-  }
-
-  requestAnimation = () => {
-    const { animationScheduled, eventsFired } = this.state;
-    if (!animationScheduled) {
-      requestAnimationFrame(this.updateWindowInfo);
-      this.setState({ animationScheduled: true, eventsFired: eventsFired + 1 });
+const WindowInfoProvider = (props) => {
+  const { children, breakpoints } = props;
+
+  const [isAnimationScheduled, setIsAnimationScheduled] = useState(false);
```

Comment:
[Refer to this review in `react-node-position`](https://github.com/trouble/react-node-position/pull/2#discussion_r359126029).